### PR TITLE
Improved engraving object tree

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

**Comment by RobFog** [comment]
Typo in commit message: "imporoved" should be "improved".

**Comment by igorkorsukov** [comment]
> Typo in commit message: "imporoved" should be "improved".

Thank you, fixed.

File: src/engraving/libmscore/volta.cpp [hunk 1]
```diff
@@ -65,8 +65,8 @@ static const ElementStyle voltaStyle {
 //   VoltaSegment
 //---------------------------------------------------------
 
-VoltaSegment::VoltaSegment(Spanner* sp, Score* s)
-    : TextLineBaseSegment(ElementType::VOLTA_SEGMENT, sp, s, ElementFlag::MOVABLE | ElementFlag::ON_STAFF | ElementFlag::SYSTEM)
+VoltaSegment::VoltaSegment(Spanner* sp, EngravingItem* parent)
```
Comment (vpereverzev) [inline src/engraving/libmscore/volta.cpp:68]: maybe it makes sense to require only one of these arguments?

like (Spanner* parent)

or (EngravingItem* parent) + explicit local check that parent is "Spanner"

Comment (igorkorsukov) [reply src/engraving/libmscore/volta.cpp:68]: You right, I will fix it

Comment (igorkorsukov) [reply src/engraving/libmscore/volta.cpp:68]: I will fix it in a next PR

**Review by vpereverzev — APPROVED** [review]

**Pull request merged**